main.tf

```diff
@@ -7,4 +7,140 @@ provider "azurerm" {
 resource "azurerm_resource_group" "project" {
   name     = "project"
   location = "West Europe"
-}
\ No newline at end of file
+}
+
+resource "azurerm_virtual_network" "hub" {
+  name                = "hub"
+  location            = azurerm_resource_group.project.location
+  resource_group_name = azurerm_resource_group.project.name
+  address_space       = [
+    "10.1.0.0/24"]
+  dns_servers         = [
+    "10.1.0.4",
+    "10.1.0.5"]
+}
+
+resource "azurerm_virtual_network" "spoke" {
+  name                = "spoke"
+  location            = azurerm_resource_group.project.location
+  resource_group_name = azurerm_resource_group.project.name
+  address_space       = [
+    "10.0.0.0/24"]
+  dns_servers         = [
+    "10.0.0.4",
+    "10.0.0.5"]
+
+  subnet {
+    name           = "vnet"
+    address_prefix = "10.0.0.0/24"
+  }
+}
+
+module "vm" {
+  source    = "./modules/ubuntu_vm"
+  comp_name = "vm2"
+  location  = azurerm_resource_group.project.location
+  nic_name  = "nic2"
+  rg_name   = azurerm_resource_group.project.name
+  admin_username = "bob"
+  admin_password = "Password1234!"
+  subnet_id = tolist(azurerm_virtual_network.spoke.subnet)[0].id
+}
+
+module "vpn_connection" {
```

Comment:
the module should be called vpn and not vpn connection because it creates the VPN and its connection